demos/benchmark/embeddings/benchmark_embeddings.py

```diff
@@ -0,0 +1,178 @@
+#
+# Copyright (c) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import asyncio
+from datasets import load_dataset, arrow_dataset , Dataset
+from tqdm.asyncio import tqdm
+import numpy as np
+import time
+import json
+import os
+import sys
+import time
+import traceback
+from dataclasses import dataclass, field
+from typing import AsyncGenerator, List, Optional, Union
+from transformers import AutoTokenizer
+import argparse
+import aiohttp
+
+AIOHTTP_TIMEOUT = aiohttp.ClientTimeout(total=6 * 60 * 60)
+
+parser = argparse.ArgumentParser(description='Run benchmark for embeddings endpoints')
+parser.add_argument('--dataset', required=False, default='Cohere/wikipedia-22-12-simple-embeddings', help='Dataset for load generation from HF or a keyword synthetic_short or synthetic_long', dest='dataset')
```

Comment:
validate possible list of datasets? display possible options, from line 42-53 i can see that there are custom ones (not from HF) - synthetic_short and synthetic_long which user may not be able to see by running --help